core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4PooledConnectionManager.java

```diff
@@ -137,12 +137,18 @@ public Connection getConnection() throws SQLException {
 				}
 			} else {
 				pc = (PooledConnection) free_.get(0);
-				if (removeFreeConnection(pc, true)) {
-					TrafT4Connection c = (TrafT4Connection) pc.getConnection();
-					try {
-						c.ic_.enforceT4ConnectionTimeout(c);
+				TrafT4Connection c = (TrafT4Connection) pc.getConnection();
+				try {
+					c.ic_.enforceT4ConnectionTimeout(c);
+					if (removeFreeConnection(pc, true)) {
 						validConnection = true;
-					} catch (SQLException sqlEx) {
+					}
+				} catch (Exception e) {
+					//ignore errors
+				} finally {
+					if (!validConnection) {
+						if (free_.remove(0) != null)
```

Comment:
@selvaganesang Thanks for you comments. i don't quit understand. It gets 0th entry first, then remove it in removeFreeConnection( here it still in 0th). If it cannot be removed, that means the connection is not in free_ which is not reasonable( unless memory issue or something else, rarely happens but doesn't matter to remove again and throw exception, then the program exit). It should be fine to remove 0th here. 
In standard doc for free_.remove: true if this list contained the specified element. it might not be represent PRESENT.